### ISSUE TITLE
Hidetsugu and Kairi: optional during-resolution free cast never reaches the player (instant or sorcery alike)

### ISSUE THREAD

**Issue by mike-theDude** [phase-rs/phase#6666]
## Summary

Hidetsugu and Kairi's dies trigger ends in an optional during-resolution free cast:

> When Hidetsugu and Kairi dies, exile the top card of your library. Target opponent loses life equal to its mana value. **If it's an instant or sorcery card, you may cast it without paying its mana cost.**

That third clause **never reaches the player**. Resolution completes to a clean `Priority` window with the exiled card still in exile and no cast offered — for an **instant and a sorcery alike**.

## Why this is a distinct defect from #518

#518 (PR #6645) fixed the *parse*: the card-type disjunction gate was dropping its `Instant` leg, lowering `"instant or sorcery"` to `RevealedHasCardType { card_types: [Sorcery] }`.

That fix is correct and necessary, but **not sufficient for this card**. The free cast fails identically whether the gate retains one leg or both, so it is downstream of the condition entirely. This was found while writing the runtime regression for #6645.

## Reproduction

`crates/engine/tests/integration/revealed_card_type_disjunction_518.rs` contains `hidetsugu_free_cast_currently_unreachable_for_either_type`, which pins the current behavior. Setup:

- Hidetsugu and Kairi on the battlefield (verbatim Oracle text)
- P0's library reduced to exactly one staged card, mana cost `{2}`
- A 0-cost destroy spell targeting Hidetsugu, cast through the real pipeline

Observed, for `top_is_instant` in `[true, false]`:

| Step | Result |
|---|---|
| Hidetsugu → graveyard | ✅ dies trigger fires (CR 700.4) |
| Top card → exile | ✅ `ExileTop` resolves |
| Opponent life | ✅ **−2**, i.e. exactly the exiled card's mana value |
| Free cast offered | ❌ **no** — card remains in `Zone::Exile` |
| `waiting_for` after resolution | `Priority { player: PlayerId(0) }` — no prompt raised |
| `legal_actions` | `[PassPriority]` |

## Notable diagnostic

The life loss lands on **exactly the exiled card's mana value**, which proves the anaphoric subject (`ObjectManaValue { scope: Anaphoric }`) binds correctly at the second clause. So the subject is available; something between there and the third clause drops it, or the optional cast is never offered.

## Parsed shape

```
ExileTop { player: Controller, count: Fixed(1) }
  -> LoseLife { amount: Ref(ObjectManaValue { scope: Anaphoric }),
                target: Typed { controller: Opponent } }
    -> CastFromZone { target: ParentTarget,
                      without_paying_mana_cost: true,
                      mode: Cast,
                      driver: DuringResolution }
       condition: RevealedHasCardType { card_types: [Instant, Sorcery] }   // post-#518
       optional: true
```

Per `crates/engine/src/game/effects/cast_from_zone.rs` (~L482), a `DuringResolution` cast is meant to put the card on the stack immediately rather than stamping a lingering permission. Neither happens here. Worth checking whether `ParentTarget` resolves to the exiled card at that point, and whether the `optional: true` offer is raised at all — no prompt appears in `waiting_for`.

Possibly related prior art: the Urza (#2880-era) and Bring to Light (#2880) work on this same `DuringResolution` routing.

## Impact

Hidetsugu and Kairi's dies trigger is half-functional in play: the exile and life loss work, the free cast silently does nothing. This is likely the behavior the original Discord reporter in #518 experienced but could not articulate.

Other cards on the `DuringResolution` + `ParentTarget` + `optional` shape may be affected; I have not swept for them.

## Not addressed by

PR #6645 — that PR fixes the parse for all 11 cards in the disjunction class and explicitly pins this defect rather than fixing it.

**Comment by mike-theDude** [comment]
## Classifier analysis

**Verdict**: `supported_aspect_defect`

**Matched clause**: `When ~ dies, exile the top card of your library. Target opponent loses life equal to its mana value. If it's an instant or sorcery card, you may cast it without paying its mana cost.` — specifically its `CastFromZone` child  (supported: true)

**Reasoning**: The card claims to support the clause. `parse_details` lowers the dies trigger to `ExileTop → LoseLife → CastFromZone`, and the `CastFromZone` node is `supported: true` with `target: parent target`, `free cast: yes`, `conditional: revealed is sorcery`. The children chain covers every sub-effect in the source text, so this is not a dropped-child collapse. Two defects sit behind that supported node. The `conditional` detail reads `revealed is sorcery` — the disjunction gate drops its `Instant` leg, which is the AST unfaithfulness already tracked by #518 and fixed by PR #6645. The behavior reported here is downstream of that: the free cast is never offered even for a sorcery, so the gate holding does not change the outcome. Inspection of `game/effects/cast_from_zone.rs` points at the anaphoric target rather than the condition — `execute_cast_from_zone` seeds `target_ids` from `ability.targets` and provides empty-target fallbacks for `ExiledBySource`, `SelfRef`, `Hand`, and `LastRevealed`, but none for `TargetFilter::ParentTarget`; an unbound anaphor leaves `target_ids` empty, takes the "No targets resolved — nothing to cast" early return, and can never satisfy the `driver_free_cast` router's `target_ids.len() == 1` requirement. That is a runtime defect, not a parser misparse.

---
_Posted by bug-coverage-classifier skill._

**Comment by mike-theDude** [comment]
## Triage analysis

**Clarified summary**
The third clause of Hidetsugu and Kairi's dies trigger — the optional free cast of the exiled card — never reaches the player. The exile and the life loss both resolve correctly; the cast is silently skipped for an instant and a sorcery alike.

**Oracle text (verified from `client/public/card-data.json`)**
> Flying
> When Hidetsugu and Kairi enters, draw three cards, then put two cards from your hand on top of your library in any order.
> When Hidetsugu and Kairi dies, exile the top card of your library. Target opponent loses life equal to its mana value. If it's an instant or sorcery card, you may cast it without paying its mana cost.

**Expected behavior**
As the dies trigger resolves, if the exiled card is an instant or a sorcery, its controller is offered the choice to cast it without paying its mana cost (CR 608.2g — the choice is made and the cast happens during resolution, not deferred to a later priority window).

**Reproduction status**
Reported behavior is pinned by `hidetsugu_free_cast_currently_unreachable_for_either_type` in `crates/engine/tests/integration/revealed_card_type_disjunction_518.rs`, which lives on the still-open PR #6645 branch, not yet on `main`. Resolution completes to `Priority { player: PlayerId(0) }` with `legal_actions == [PassPriority]` and the card still in `Zone::Exile`, for `top_is_instant` in `[true, false]`.

**Investigation findings**
Coverage data confirms the card *claims* to support the clause: the dies trigger lowers to `ExileTop → LoseLife → CastFromZone`, and the `CastFromZone` child is `supported: true` with details `target: parent target`, `free cast: yes`, `conditional: revealed is sorcery`. So the reported symptom is not an unsupported-gap; it is a defect behind a supported node.

Two distinct problems sit on that node, and only one of them is #518's:

1. *(tracked by #518 / PR #6645)* The published condition reads `revealed is sorcery` — the disjunction gate drops its `Instant` leg. PR #6645 fixes the parse for all 11 cards in that class.
2. *(this issue)* The cast fails identically once the gate holds, so it is downstream of the condition. Code inspection points at target binding rather than the condition or the driver: `execute_cast_from_zone` (`crates/engine/src/game/effects/cast_from_zone.rs`) seeds `target_ids` from `ability.targets`, then supplies empty-target fallbacks for `ExiledBySource`, `SelfRef`, `Hand`, and `LastRevealed` filters — but there is no fallback for `TargetFilter::ParentTarget`. If the anaphoric "it" is not seeded with the card `ExileTop` just exiled, `target_ids` stays empty and the handler takes its "No targets resolved — nothing to cast" early return, which emits `EffectResolved { subject: None }` and returns. That is a silent no-op with no prompt raised — exactly the observed shape. The `driver_free_cast` router below it additionally requires `target_ids.len() == 1`, so an unbound anaphor can never reach the during-resolution cast path even though the `driver: DuringResolution` discriminator is correct.

Consistent with that reading, the life-loss step lands on exactly the exiled card's mana value, proving the anaphoric subject binds at the *second* clause (`ObjectManaValue { scope: Anaphoric }`) — so the referent exists in the resolution context and the gap is in how the third clause's `ParentTarget` filter reads it. Two candidate fix shapes worth weighing: seed the `ParentTarget` slot from the exiling sub-effect's published id, or lower "it" to a filter the handler already has an empty-target fallback for. This is an inspection-level lead, not a proven root cause.

**Likely scope**
`area:engine` — `game/effects/cast_from_zone.rs` (target seeding and the `driver_free_cast` router), plus the chained-anaphor seeding in `game/ability_utils.rs` / `game/stack.rs`. Cards on the same `DuringResolution` + `ParentTarget` + `optional` shape are plausibly affected as a class; that sweep has not been done.

**Acceptance criteria**
- Resolving the dies trigger with an instant on top offers the controller the free cast during resolution, and accepting puts the card on the stack rather than leaving it in exile.
- The same holds with a sorcery on top.
- Declining leaves the card in exile with no lingering cast permission stamped.
- The exile and life-loss steps are unchanged.
- A runtime test drives the real cast → stack → dies-trigger pipeline and is proven discriminating against the current behavior.

**Related issues**
- #518 / PR #6645 — the parse-level `instant or sorcery` gate fix for this same card. Necessary but not sufficient; PR #6645 explicitly pins this defect rather than fixing it. Not a duplicate.
- #6356 (Rashmi and Ragavan — "correctly exiles and even highlights like it will let you cast but never does") is the closest sibling: same user-visible shape, an exiled card whose conditional optional free cast is never delivered. Kept separately actionable — different card, different condition, and Rashmi's clause additionally carries an "if you don't cast it this way, you may cast it this turn" lingering-permission leg that this card has no analogue for. Worth checking whether one fix closes both.
- #3674 (Mosswort Bridge, closed) and #2880-era Bring to Light / Urza work are prior art on the `DuringResolution` routing.

**Comment by ntindle** [comment]
## Bug-state backfill evidence — #6666 (Hidetsugu and Kairi optional free cast)

**Update 2026-09-17 — still `reproduced` on engine v0.85.0** (`cb58ef5`, WS protocol 72). Instant leg: Mystical Tutor stages Lightning Bolt on top, then Murder on H&K.

**What's new vs the 2026-09-10 run:** the optional free-cast prompt now reaches the player — the engine issues an `OptionalEffectChoice` with accept/decline choices, and the accept submission is applied cleanly (no rejection). But accepting does nothing: the trigger resolves with the staged Lightning Bolt stranded in Exile, P1 at 19 life (only the 1-life dies-trigger drain; a resolved free Bolt would put P1 at 16), no cast or target prompt ever appears, and the game proceeds normally. Same reported outcome — the free cast never reaches the player — on the current pinned release.

**Assertions** (instant leg; failures = the reproduced bug):
- A1 setup ok, A2 H&K on battlefield, A3 top staged, A4 H&K died, A5 exile + life loss = MV: **passed**
- A6 free-cast offer reaches P0: **passed**
- A7 accepted cast resolves fully: **failed** — Bolt in Exile post-trigger, P1 life 19 (expected 16)
- A8 end state clean (`Priority`, empty stack, no dangling decision): **passed**

**Provenance & limitations:** prebuilt `phase-server` v0.85.0 release binary (minisign-verified with the repo-pinned key; card-data.json + draft-pools.json SHA-verified against the release manifest). States captured via `ExportAuthoritativeState` (PlayerId 0, single-user mode). Browser UI not exercised; native engine via two human-client seats. Saved states are observed snapshots, restorable only by replaying the driver scenario.

**Evidence** (commit-pinned, immutable; 9 files, all byte-verified after upload, all JSON parse, PNG readable):
- [summary.png](https://raw.githubusercontent.com/ntindle/phase-bug-state-evidence/70d5c2c532e01a91e4bfc7b6a5d529088fe75c9a/6666/20260917-6666/summary.png)
- [run.json](https://raw.githubusercontent.com/ntindle/phase-bug-state-evidence/70d5c2c532e01a91e4bfc7b6a5d529088fe75c9a/6666/20260917-6666/run.json)
- [instant_pre_trigger](https://raw.githubusercontent.com/ntindle/phase-bug-state-evidence/70d5c2c532e01a91e4bfc7b6a5d529088fe75c9a/6666/20260917-6666/instant_pre_trigger.json) / [instant_mid_tutor](https://raw.githubusercontent.com/ntindle/phase-bug-state-evidence/70d5c2c532e01a91e4bfc7b6a5d529088fe75c9a/6666/20260917-6666/instant_mid_tutor_1789642069.json) / [instant_post_trigger](https://raw.githubusercontent.com/ntindle/phase-bug-state-evidence/70d5c2c532e01a91e4bfc7b6a5d529088fe75c9a/6666/20260917-6666/instant_post_trigger.json) / [instant_post](https://raw.githubusercontent.com/ntindle/phase-bug-state-evidence/70d5c2c532e01a91e4bfc7b6a5d529088fe75c9a/6666/20260917-6666/instant_post.json)
- [wire_log.jsonl](https://raw.githubusercontent.com/ntindle/phase-bug-state-evidence/70d5c2c532e01a91e4bfc7b6a5d529088fe75c9a/6666/20260917-6666/wire_log.jsonl) / [scenario_run.log](https://raw.githubusercontent.com/ntindle/phase-bug-state-evidence/70d5c2c532e01a91e4bfc7b6a5d529088fe75c9a/6666/20260917-6666/scenario_run.log) / [manifest.sha256](https://raw.githubusercontent.com/ntindle/phase-bug-state-evidence/70d5c2c532e01a91e4bfc7b6a5d529088fe75c9a/6666/20260917-6666/manifest.sha256)

**Prior run (2026-09-10, v0.78.0):** `reproduced` — the prompt never reached the player then (no OptionalEffectChoice at all); the exiled card stayed in exile and only the life-loss applied. Evidence: [summary.png](https://raw.githubusercontent.com/ntindle/phase-bug-state-evidence/34cb0e1e16877e4e2e9b9d98d1183321125b2518/6666/20260910-6666/summary.png) / [run.json](https://raw.githubusercontent.com/ntindle/phase-bug-state-evidence/34cb0e1e16877e4e2e9b9d98d1183321125b2518/6666/20260910-6666/run.json) / [states](https://raw.githubusercontent.com/ntindle/phase-bug-state-evidence/34cb0e1e16877e4e2e9b9d98d1183321125b2518/6666/20260910-6666/A_post_trigger.json) / [wire_log](https://raw.githubusercontent.com/ntindle/phase-bug-state-evidence/34cb0e1e16877e4e2e9b9d98d1183321125b2518/6666/20260910-6666/wire_log.jsonl)
